vault/src/main.rs

```diff
@@ -13,9 +20,82 @@ use vault::{
     VaultService, VaultServiceConfig, ABOUT, AUTHORS, NAME, VERSION,
 };
 
+#[derive(Parser)]
+#[clap(args_conflicts_with_subcommands = true)]
+struct Cli {
+    #[clap(subcommand)]
+    sub: Option<Commands>,
+
+    #[clap(flatten)]
+    opts: RunVaultOpts,
+}
+
+#[derive(Parser)]
+enum Commands {
+    GenerateWif(GenerateWifOpts),
+    GenerateKey(GenerateKeyOpts),
+    #[clap(name = "run")]
+    RunVault(RunVaultOpts),
+}
+
+#[derive(Debug, Parser, Clone)]
+struct GenerateWifOpts {
+    /// Output file name or stdout if unspecified.
+    #[clap(parse(from_os_str))]
+    output: Option<PathBuf>,
+
+    #[clap(long)]
+    network: Network,
+}
+
+impl GenerateWifOpts {
+    fn generate_and_write(&self) -> Result<(), Error> {
+        let secret_key = SecretKey::new(&mut thread_rng());
+        let private_key = PrivateKey::new(secret_key, self.network);
+        let wif = private_key.to_wif();
+        let data = wif.as_bytes();
+
+        if let Some(output) = &self.output {
+            std::fs::write(output, data)?;
+        } else {
+            std::io::stdout().write_all(data)?;
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser, Clone)]
+struct GenerateKeyOpts {
+    /// Output file name or stdout if unspecified.
+    #[clap(parse(from_os_str))]
+    output: Option<PathBuf>,
+}
+
+impl GenerateKeyOpts {
+    fn generate_and_write(&self) -> Result<(), Error> {
+        let (pair, phrase, _) = KeyPair::generate_with_phrase(None);
+
+        let mut keys = serde_json::Map::new();
+        keys.insert(
+            pair.public().to_ss58check_with_version(SS58_PREFIX.into()),
+            serde_json::Value::String(phrase),
+        );
+        let data = serde_json::to_vec(&keys)?;
+
+        if let Some(output) = &self.output {
+            std::fs::write(output, data)?;
```

Comment:
Would this overwrite existing files? I think it's better to return an error rather than overwriting